### PR TITLE
Nil handler bug fix

### LINKED ISSUE #100: Panic when route with `nil` handler matched
If you use a registration call, like `Router.Handle`, and provide `nil` in place of a handler, Matcha panics when the given route is matched. While this is sort of an edge case, ideally Matcha wouldn't panic in most circumstances, and this is a very preventable one. So, to fix this issue:

- Determine where a registered `nil` handler is called
- Add a guard (`if handler != nil...`) and only call the handler if it isn't nil
- Determine an appropriate default response code if the handler is nil (no body)
- Test and document the new behavior

Please don't solve this issue by throwing an error if a user provides a nil handler; this is potentially valuable behavior since it lets users safely create placeholder routes to implement later.

### PR DESCRIPTION
**Pull request by aritt0923**
#100 
Matcha no longer panics when route with 'nil' handler is matched, instead throws HTTP status code 501, Not Implemented. Test added to include coverage for this edge case.

### CODE REVIEW

**Comment by jakenichols2719** [comment]
Thanks for your pull request, aritt! Apologies for the CI failure; when the org name changed, I had to update to a new Coveralls project, and forgot to re-set the values for failure threshold. It was set to fail on *any* test coverage drop. This has now been fixed, so your next commit should pass as long as tests pass.

I'm adding some additional comments in code review--please give them a look when you're able!

File: .vscode/launch.json
Comment (jakenichols2719) [inline .vscode/launch.json:1]: Please make sure to remove `.vscode` artifacts from your commits. Honestly, it should be in .gitignore, and I'm not sure why it isn't... either way, please `git rm` this file.

File: pkg/router/config.go
Comment (jakenichols2719) [inline pkg/router/config.go:1]: Good catch, thank you!

File: pkg/router/router_test.go [hunk 1]
```diff
@@ -224,6 +225,7 @@ func TestEdgeCaseRoutes(t *testing.T) {
 		WithRoute(route.Declare(http.MethodGet, "/odd///path"), okHandler("odd")),
 		HandleRoute(route.Declare(http.MethodGet, "/reject", route.WithMiddleware(reject)), okHandler("never")),
 	)
+	r.HandleFunc(http.MethodGet, "/not/implemented", nil)
```
Comment (jakenichols2719) [inline pkg/router/router_test.go:228]: I know it's a total pain, but for this specific case the behavior should be tested with all four handling signatures (`Handle`/`HandleFunc`, `HandleRoute`/`HandleRouteFunc`). This should resolve the coverage and also ensure we don't get caught by surprise by this again.

**Comment by jakenichols2719** [comment]
Thanks for the updates! Merging now.
